DA1469x clock control driver

blauret opened the pull request:
This commit adds basic support for the clock controller used in SmartBond Da1469x MCUs family

Signed-off-by: Ben Lauret <ben.lauret.wm@renesas.com>

kasjer committed:
drivers: clock_control: smartbond: initial support

This commit adds basic support for the clock controller used in
SmartBond MCUs.

Signed-off-by: Jerzy Kasenberg <jerzy.kasenberg@codecoup.pl>

galak commented on dts/bindings/clock/renesas,smartbond-clock.yaml:
```diff
@@ -0,0 +1,12 @@
+
+properties:
+  reg:
+    required: false
```
remove `required: false` its the default.

blauret replied:
Done

galak reviewed (commented):
How do you expect the request/release api's to be used?

blauret commented:
> How do you expect the request/release api's to be used?

There are multiple use cases. The PLL is required to run the USB interface (PR is coming for the USB). It will request the PLL and as long as the USB is enabled it keeps it running. If you disable the USB it allows the system to fall back to the 32MHz. 

The other use case is for application requiring more computing power. The application can request the PLL for faster processing. It's specificaly useful in conjunction with the LCD controller. An example is when you get an interruption to turn on the screen, application requests the PLL for faster update. When the screen turns off it is released to save power.

kasjer committed:
boards: DA1469x Pro DK enable oscillator in DT

This enables XTAL32M XTAL32K and PLL on DA1469x Pro DK
so USB and BT can be used without additional overly.

Signed-off-by: Jerzy Kasenberg <jerzy.kasenberg@codecoup.pl>

ydamigos commented on drivers/clock_control/clock_control_smartbond.c:
```diff
@@ -0,0 +1,288 @@
+struct clocks_dev_data {
+	/* RCX rate, can be measured at runtime */
+	uint32_t rcx_rate;
+	/* RCX rate, can be measured at runtime */
```
RCX -> RC32K

ydamigos commented on dts/arm/renesas/smartbond/da1469x.dtsi:
```diff
@@ -19,6 +20,66 @@
+				compatible = "fixed-clock";
+				clock-frequency = <DT_FREQ_M(96)>;
+				#clock-cells = <0>;
+				status = "disabled";
```
We should state the dependency between PLL and XTAL32M. Right now if xtal32m status is disabled and pll status is ok, we could enable PLL in `smartbond_clocks_init()` and disable XTAL32M.

ydamigos commented on drivers/clock_control/clock_control_smartbond.c:
```diff
@@ -0,0 +1,288 @@
+		*rate = data->rcx_rate;
+		break;
+	case DT_DEP_ORD(DT_NODELABEL(xtal32k)):
+		*rate = DT_PROP(DT_NODELABEL(xtal32k), clock_frequency);
```
It would be good to configure XTAL32K before enabling it.

MaureenHelm commented:
@blauret ping

kasjer commented:
Clock control code was update notable differences to previous iteration:
- renesas/hal (from separate repository) is used for actual xtal/rc/pll manipulation
- PLL request functions were removed and now PLL is enabled in DT
- additional static assert is added to prevent case when PLL is enabled but XTAL32M is not (noticed by @ydamigos)

blauret commented:
@MaureenHelm, thanks for checking on this one. We updated the PR and we believe all prior comments have been addressed
